### DISCUSSION TITLE
Prevent settings reset after firmware update

### DISCUSSION THREAD

**Discussion by sn0opy** [Jyers/Marlin#427] [Q&A] [answered]
I'm not entirely sure when this started, but after the most recent Smith3D firmware updates, I experienced the same issue. Everytime I update the firmware, settings are being reset to defaults which means, you have to re-configure e-steps, probe offset and z-offset at least.

Is there any way to prevent this from happening?

**Comment by aryajuanda** [comment]
compile it with your personal setting ... 
i always do manual compile with my probe to nozzle offset, probe margin, temperature for preheat filament, filament type, and some other personal config, so after flash firmware "its just work" ...

**Comment by Jyers** [comment, marked as answer]
It is best practice to reset eeprom after flashing anyway to avoid errors. @aryajuanda is correct that is you compile your own you can set these parameters as default. Besides this there is no good way of avoiding this besides perhaps functionality being added such as #420

**Comment by t3chguy** [comment]
This is not really an issue for this repo but one for upstream Marlin if anything.

**Reply by sn0opy** [reply]
Never said it was an issue, but an improvement, that could be added to this fork. I'd say #420 is more fitting and the actual thing that would be a good thing to implement since I don't want/need my personal build only to keep 4 settings.

**Reply by t3chguy** [reply]
in either case it'd just increase the difficulty in getting this upstreamed into stable Marlin
